File: frontend/src/types/location.ts
/** Domain types for gym locations/branches. Mirrors backend/app/schemas/location.py. */

export interface Location {
  id: number;
  name: string;
  address: string | null;
  phone: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

/** Payload for `POST /api/v1/locations`. */
export interface LocationCreatePayload {
  name: string;
  address?: string | null;
  phone?: string | null;
}

/** Payload for `PUT /api/v1/locations/{id}` — all fields optional. */
export type LocationUpdatePayload = Partial<LocationCreatePayload> & { is_active?: boolean };

File: backend/app/schemas/location.py
"""Pydantic schemas for gym locations/branches."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class LocationCreate(BaseModel):
    """Payload to create a new location/branch."""

    name: str = Field(..., min_length=1, max_length=150)
    address: str | None = Field(default=None, max_length=500)
    phone: str | None = Field(default=None, max_length=30)


class LocationUpdate(BaseModel):
    """Payload to update an existing location. All fields optional."""

    name: str | None = Field(default=None, min_length=1, max_length=150)
    address: str | None = Field(default=None, max_length=500)
    phone: str | None = Field(default=None, max_length=30)
    is_active: bool | None = None


class LocationResponse(BaseModel):
    """A location/branch as returned by the API."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    address: str | None
    phone: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime
